instrumentation/spdlog/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.12)
+
+set(this_target opentelemetry_spdlog_sink)
+
+project(${this_target})
+
+find_package(opentelemetry-cpp REQUIRED)
+find_package(spdlog REQUIRED)
+
+add_library(${this_target} src/sink.cc)
+
+target_compile_options(${this_target} PUBLIC
+  -Wall -Wextra -Werror -Wpedantic -fPIC
+)
+
+set_target_properties(${this_target} PROPERTIES EXPORT_NAME ${this_target})
+
+target_include_directories(${this_target} PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+  ${OPENTELEMETRY_CPP_INCLUDE_DIRS}
+)
+
+target_link_libraries(${this_target} PRIVATE
+  spdlog::spdlog
+)
+
+if(OPENTELEMETRY_INSTALL)
+  install(
+    TARGETS ${this_target}
+    EXPORT "${PROJECT_NAME}-target"
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
```

Comment:
Probably also makes sense to export the target itself so that it's easier to use this library in another cmake-based project, smth like

```
  install(
    EXPORT "${PROJECT_NAME}-target"
    FILE "${PROJECT_NAME}-target.cmake"
    NAMESPACE "${PROJECT_NAME}::"
    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
```
?

Ideally also a config (see e.g. https://cmake.org/cmake/help/latest/guide/importing-exporting/index.html#creating-a-package-configuration-file) for easier discoverability / reuse?

Reply:
That's reasonable, I'll have a look. I was seeing it as a Makefile-based build only.